app/src/main/java/edu/wisc/ece/pinpoint/pages/pins/PinViewFragment.java

```diff
@@ -68,11 +90,51 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
         textContent = requireView().findViewById(R.id.pin_text_content);
         imageContent = requireView().findViewById(R.id.pin_image_content);
         metadataBar = requireView().findViewById(R.id.pin_view_metadata);
-        ImageButton backButton = requireView().findViewById(R.id.pin_view_back_button);
-        ImageButton optionsButton = requireView().findViewById(R.id.pin_view_options_button);
+        backButton = requireView().findViewById(R.id.pin_view_back_button);
+        addCommentButton = requireView().findViewById(R.id.add_comment_button);
+        sendCommentButton = requireView().findViewById(R.id.send_comment_button);
+        addCommentEditText = requireView().findViewById(R.id.comment_edittext_layout);
+        addCommentInputLayout = requireView().findViewById(R.id.comment_input_layout);
+        scrollView = requireView().findViewById(R.id.viewpin_scrollview);
 
         backButton.setOnClickListener((v) -> navController.popBackStack());
-        optionsButton.setOnClickListener(this::showOptionsMenu);
+
+        addCommentButton.setOnClickListener((v) -> {
+            if(addCommentInputLayout.getVisibility() == View.GONE){
+                addCommentButton.setForeground(ContextCompat.getDrawable(getContext(), R.drawable.ic_cancel_comment));
+                addCommentInputLayout.setVisibility(View.VISIBLE);
+                sendCommentButton.setVisibility(View.VISIBLE);
+            }
+            else{
+                addCommentButton.setForeground(ContextCompat.getDrawable(getContext(), R.drawable.ic_add_comment));
+                addCommentInputLayout.setVisibility(View.GONE);
+                sendCommentButton.setVisibility(View.GONE);
+            }
+        });
+
+        sendCommentButton.setOnClickListener((v) -> {
+            if (addCommentEditText.getText() != null) {
+                Comment comment = new Comment(addCommentEditText.getText().toString());
+                firebase.postComment(comment, pid);
+
+                // clear comment text and hide posting components
+                addCommentEditText.setText("");
+                addCommentButton.setForeground(ContextCompat.getDrawable(getContext(), R.drawable.ic_add_comment));
```

Comment:
Once this is in the `onSuccessListener` it should be wrapped in an `if (getContext() != null)` check to ensure the app doesn't crash if the user navigates away from the page.